File: Backend/src/main/java/com/feedle/feedleapi/Networking/GetPostsResponse.java
package com.feedle.feedleapi.Networking;

import com.feedle.feedleapi.Models.Post;

import java.util.List;

public class GetPostsResponse extends Request {
    private List<Post> posts;

    public GetPostsResponse(List<Post> posts) {
        super(RequestType.GetPosts);
        this.posts = posts;
    }

    public List<Post> getPostList() {
        return posts;
    }
}

File: FeedleDataTier/FeedleDataTier/Network/GetPostsResponse.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;
using FeedleDataTier.Models;

namespace FeedleDataTier.Network
{
    public class GetPostsResponse : Request
    {
        [JsonPropertyName("posts")]
        public List<Post> Posts { get; set; }
        
        public GetPostsResponse(List<Post> posts) : base(RequestType.GetPosts)
        {
            this.Posts = posts;
        }
    }
}